### DISCUSSION TITLE
How to define a function similar to con*() to validate an integer only from within a given set of numbers

### DISCUSSION THREAD

**Discussion by leonixyz** [pydantic/pydantic#3164] [Question] [answered]
I would like to handle something similar to:

    class MyModel(BaseModel):
        p: one_of([1,5,1000])

Is there a way to do it?
I want my `p` to be an int but only from within that list.

What is the most concise way to achieve it?

Thanks

**Comment by nymous** [comment, marked as answer]
Hello @leonixyz, does [`Literal`](https://mypy.readthedocs.io/en/stable/literal_types.html) meet your needs?
```py
from typing import Literal

from pydantic import BaseModel


class MyModel(BaseModel):
    p: Literal[1, 5, 1000]


MyModel(p=1)  # OK
MyModel(p=10)
# pydantic.error_wrappers.ValidationError: 1 validation error for MyModel
# p
#   unexpected value; permitted: 1, 5, 1000 (type=value_error.const; given=10; permitted=(1, 5, 1000))
```

**Reply by leonixyz** [reply]
Thank you. This is exactly what I wanted. And it also works with strings that allows me to avoid defining enumerators.
Thanks!
